unlock-app/src/reducers/locksReducer.js

```diff
@@ -81,6 +82,16 @@ const locksReducer = (state = initialState, action) => {
     }
   }
 
+  if (action.type === DELETE_LOCK) {
+    if (!state[action.address]) {
```

Comment:
Errors should not be thrown in a reducer. The error condition should be detected either at user input, or if input could come asynchronously from web3, then in a middleware, which would then dispatch an error action.

Reply:
Hum. Do you have any more data on that?
This is also a pattern used all over the place and that worked fine?
My worry is that we obviously do not want to check the state for every action triggered (this would duplicate the effort... and, more importantly means that anyone triggering an action should know the state before which seems like an anti-pattern.

Reply:
Now that I think about it, I will actually delete all these errors because they are actually never triggered. They are more like `asserts` in a way. I'll submit a PR which removes them all.